Tools/autotest/common.py

```diff
@@ -456,8 +458,8 @@ def do_sport_read(self):
                 if b != self.SPORT_DATA_FRAME:
                     # we may come into a stream mid-way, so we can't judge
                     self.progress("############# Bad char %x" % b)
-                    raise ValueError("Bad char (0x%02x)" % b)
```

Comment:
Eep!  major function change.  Please revert this one and fix whatever needed fixing in some other way :-)

Removing bad_chars entirely is an option.

Reply:
Ok so raising the exception was the good way?
I will revert and do the same for the other case